server/events.py

```diff
@@ -204,6 +220,25 @@ def __call__(self, request: EventRequest, room_code: str) -> tuple[bool, Client
                 )
                 WebSocketDisconnect.response = response
                 raise WebSocketDisconnect
+            case EventType.SYNC:
+                # Send the sync event to the client to update code/collaborators
+                # Send an event to everyone else to update collaborators
+                data = cast(SyncData, request.data)
+                await self.client.send(EventResponse(type=EventType.SYNC, data=data, status_code=StatusCode.SUCCESS))
+                connect_data = cast(
+                    ConnectData,
+                    {
+                        "connection_type": "join",
+                        "difficulty": None,
+                        "room_code": room_code,
+                        "username": self.client.username,
+                    },
+                )
```

Comment:
The `cast` method shouldn't be required, the data could be represented like so:
```py
ConnectData(connection_type="join", room_code=room_code, username=self.client.username)
```
which, in my opinion, looks more clear and avoids typing errors.

Reply:
Type checkers don't like this which is why I casted the EventData to each of their actual types. The casts also provided me with intellisense. The cast is only used to appease the type checkers